test/test_xcpp_kernel.py

```diff
@@ -132,6 +133,54 @@ class XCppTests(jupyter_kernel_test.KernelTests):
             }
         ]
 
+    #Helper function to execute the notebooks
+    def execute_notebook(inp, out):
+
+        with open(inp) as f:
+            input_nb = nbformat.read(f, as_version=4)
+
+        # Execute the notebook
+        executed_notebook = pm.execute_notebook(
+            inp,
+            out,
+            log_output=True,
+            kernel_name='xcpp20'
+        )
+
+        with open(out) as f:
+            output_nb = nbformat.read(f, as_version=4)
+
+        check = True
+
+        # Iterate over the cells in the input and output notebooks
+        for input_cell, output_cell in zip(input_nb.cells, output_nb.cells):
+            if input_cell.cell_type == 'code' and output_cell.cell_type == 'code':
+                # If one cell has output and the other doesn't, set check to False
+                if bool(input_cell.outputs) != bool(output_cell.outputs):
+                    check = False
+                    break
+                else:
+                    for input_output, output_output in zip(input_cell.outputs, output_cell.outputs):
+                        if input_output.get('text') != output_output.get('text'):
+                            check = False
+                            break
+        
+        return check
+
+    # Tests for Notebooks
+    class XCppNotebookTests(unittest.TestCase):
+
+        notebook_pairs = [
+        ('Notebooks/test.ipynb', 'Notebooks/test_output.ipynb'),
```

Comment:
We do not need to force the user to write `test_output.ipynb` as this is an implementation detail on how we process the notebook. We can copy with a fixed prefix.